### PR TITLE
KCFG viewer tweaks

### PR DESCRIPTION
**Pull request by goodlyrottenapple**
Hide custom view when none is supplied and show constraints in the tree view as booleans not ML. 

Note: These are being added to make the `kimp` demo tool more user friendly. See the original:

image

vs tweaks to pyk + better pretty printing in the LHS tree viewer:

image

@tothtamas28 the one thing I don't see a way to remove is the toggle custom key binding, as that seems to be a class constant. Would you know of a way to add it conditionally?

### CODE REVIEW

**Comment by tothtamas28** [comment]
> @tothtamas28 the one thing I don't see a way to remove is the toggle custom key binding, as that seems to be a class constant. Would you know of a way to add it conditionally?

This can be made a regular object-level attribute / method if necessary.

**Review by tothtamas28 — COMMENTED** [review]
Looks good to me. The only thing I'm a bit concerned about is how it affects downstream projects (`evm-semantics`, `kontrol`). Can you check preemptively?

**Comment by goodlyrottenapple** [comment]
> Looks good to me. The only thing I'm a bit concerned about is how it affects downstream projects (`evm-semantics`, `kontrol`). Can you check preemptively?

I used this branch of pyk to test and display some proofs in kevm and didn't observe any issues...

**Review by tothtamas28 — APPROVED** [review]

**Review by tothtamas28 — APPROVED** [review]

File: src/pyk/kcfg/show.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def _print_split_edge(split: KCFG.Split, target_id: int) -> list[str]:
             csubst = split.splits[target_id]
             ret_split_lines: list[str] = []
             substs = csubst.subst.items()
-            constraints = csubst.constraints
+            constraints = [ml_pred_to_bool(c, unsafe=True) for c in csubst.constraints]
```
Comment (ehildenb) [inline src/pyk/kcfg/show.py:162]: This change will also upset a lot of output in downstream projects. We should make it optional at the very least.

**Pull request merged**